### PR TITLE
Sql examples 2

### PR DESCRIPTION
**Pull request by tvv923**
Sql examples 2

### CODE REVIEW

File: jp/src/edu/hillel/lesson21/DataBaseConnection.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public Connection getConnection() {
+        try {
+            return DriverManager.getConnection(URL, USER, PASSWORD);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public void close(Connection connection) {
```
Comment (kartius) [inline jp/src/edu/hillel/lesson21/DataBaseConnection.java]: Please, implement AutoClosable interface by this class and override close method to close connection properly

File: jp/src/edu/hillel/lesson21/LessonDao.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        try (PreparedStatement statement = connection.prepareStatement(query);
+             ResultSet resultSet = statement.executeQuery()) {
+            while (resultSet.next()) {
+                int lessonId = resultSet.getInt("id");
+                String lessonName = resultSet.getString("name");
+                int homeworkId = resultSet.getInt("homework_id");
+                String homeworkName = resultSet.getString("homework_name");
+                String homeworkDescription = resultSet.getString("homework_description");
+
+                Homework homework = new Homework(homeworkName, homeworkDescription);
```
Comment (kartius) [inline jp/src/edu/hillel/lesson21/LessonDao.java]: I recommend to use additional constructor to set id from database

File: jp/src/edu/hillel/lesson21/LessonDao.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                homework.setId(homeworkId);
+
+                Lesson lesson = new Lesson(lessonName, homework);
```
Comment (kartius) [inline jp/src/edu/hillel/lesson21/LessonDao.java]: And the same here

**Review by kartius — APPROVED** [review]

File: jp/src/edu/hillel/lesson21/LessonDao.java (hunk 3 above)
Comment (tvv923) [reply jp/src/edu/hillel/lesson21/LessonDao.java]: Changed

File: jp/src/edu/hillel/lesson21/LessonDao.java (hunk 2 above)
Comment (tvv923) [reply jp/src/edu/hillel/lesson21/LessonDao.java]: Changed

File: jp/src/edu/hillel/lesson21/DataBaseConnection.java (hunk 1 above)
Comment (tvv923) [reply jp/src/edu/hillel/lesson21/DataBaseConnection.java]: Changed

**Review by kartius — APPROVED** [review]

**Pull request merged**